### DISCUSSION TITLE
Firmata support

### DISCUSSION THREAD

**Discussion by eadmaster** [BrianPugh/belay#162] [Ideas]
it would make this library usable with boards not supporting micropython (old arduinos, stm32, etc)

It would also make a nice replacement for pyFirmata and pyMata that are mostly discontinued at this point.

I understand some features  prolly wont be usable, like the package manager.

**Comment by BrianPugh** [comment]
Unfortunately, it's a hard requirement that micropython is running on the target microcontroller. Belay doesn't have an explicit device firmware like firmata. Belay works by sending over python code that gets executed in micropython, so it's very fundamental to the functionality of Belay.
